var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -20,11 +20,11 @@ class Julia(Package):
     # depends_on("pkg-config", deptypes='build')
 
     # Combined build-time and run-time dependencies:
-    depends_on("binutils", deptypes='build')
-    depends_on("cmake @2.8:", deptypes='build')
-    depends_on("git")
-    depends_on("openssl")
-    depends_on("python @2.7:2.999")
+    depends_on("binutils", deptypes=('build', 'run'))
+    depends_on("cmake @2.8:", deptypes=('build', 'run'))
+    depends_on("git", deptypes=('build', 'run'))
+    depends_on("openssl", deptypes=('build', 'run'))
+    depends_on("python @2.7:2.999", deptypes=('build', 'run'))
```

Comment:
The package manager links against them. However, from Spack's point of view, the package manager runs at run-time.

Yes, I added these so that the package manager can find them. For HDF5, the package manager could also install its own version of HDF5 (but why should it). For MPI, integrating with a system MPI seems prudent.

The latter two (HDF5 and MPI) should probably be variants, enabled by default. I can make this change later.